File: JAVA/Pertemuan 01/src/id/ac/uph/pbo/pertemuan01/Ponsel.java
package id.ac.uph.pbo.pertemuan01;

public class Ponsel {
    private String merek;
    private String model;
    private double ukuranLayar;   // inch
    private int    kapasitasBaterai; // mAh
    private int    bateraiPersen;

    public Ponsel(String merek, String model, double ukuranLayar, int kapasitasBaterai) {
        this.merek            = merek;
        this.model            = model;
        this.ukuranLayar      = ukuranLayar;
        this.kapasitasBaterai = kapasitasBaterai;
        this.bateraiPersen    = 100;
    }

    public void telepon(String nomor) {
        System.out.println("[" + merek + " " + model + "] Menelepon: " + nomor);
    }

    public void ambilFoto() {
        System.out.println("[" + merek + " " + model + "] Foto diambil 📸");
    }

    public void cas() {
        bateraiPersen = 100;
        System.out.println("[" + merek + " " + model + "] Baterai penuh: 100%");
    }

    public void tampilkanInfo() {
        System.out.println("  Ponsel    : " + merek + " " + model);
        System.out.println("  Layar     : " + ukuranLayar + " inch");
        System.out.println("  Baterai   : " + kapasitasBaterai + " mAh (" + bateraiPersen + "%)");
    }
}

File: PYTHON/Pertemuan 01/Ponsel.py
import sys
sys.stdout.reconfigure(encoding='utf-8')
class Ponsel:
    def __init__(self, merek: str, model: str, ukuran_layar: float, kapasitas_baterai: int):
        self.merek = merek
        self.model = model
        self.ukuran_layar = ukuran_layar
        self.kapasitas_baterai = kapasitas_baterai
        self.baterai_persen = 100

    def telepon(self, nomor: str):
        print(f"[{self.merek} {self.model}] Menelepon: {nomor}")

    def ambil_foto(self):
        print(f"[{self.merek} {self.model}] Foto diambil 📸")

    def cas(self):
        self.baterai_persen = 100
        print(f"[{self.merek} {self.model}] Baterai penuh: 100%")

    def tampilkan_info(self):
        print(f"  Ponsel    : {self.merek} {self.model}")
        print(f"  Layar     : {self.ukuran_layar} inch")
        print(f"  Baterai   : {self.kapasitas_baterai} mAh ({self.baterai_persen}%)")

